jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -193,56 +187,65 @@ YUI.add('user-profile-model-list', function() {
       const uuid = model.uuid;
       // Hide the confirmation popup.
       this._displayConfirmation(null);
-      const xhr = this.props.destroyModels([uuid], this._destroyModelCallback);
-      this.xhrs.push(xhr);
+      // Add the model to the list of models being destroyed, then make the
+      // actual API call. The API call is in a callback to avoid race
+      // conditions.
+      this.setState({
+        destroyingModels: this.state.destroyingModels.concat([uuid])
+      }, () => {
+        const callback = this._destroyModelCallback.bind(
+          this, uuid, model.name);
+        this.props.destroyModels([uuid], callback);
+      });
     },
 
     /**
       Callback for the controller delete model call.
 
       @method _destroyModelCallback
+      @param {String} uuid the model being deleted. Bound when the callback is
+        passed into the API call.
+      @param {String} modelName the name of the model being deleted; primarily
+        useful for providing user-friendly notifications messages. Bound when
+        the callback is passed into the API call.
       @param {String} err The error from the request, or null.
-      @param {Array} modelList The list of models.
+      @param {Object} results The result for the model being deleted. The
+        object is keyed to the model UUID.
     */
-    _destroyModelCallback: function(err, results) {
+    _destroyModelCallback: function(uuid, modelName, err, results) {
       const addNotification = this.props.addNotification;
-      // Handle global errors.
-      if (err) {
+      // Handle global errors or model-specific errors.
+      const error = err || results[uuid];
+      if (error) {
         addNotification({
           title: 'Model destruction failed',
-          message: 'The model failed to be destroyed: ' + err,
+          message: `Could not destroy model "${modelName}": ${error}`,
```

Comment:
Decided to make the messages here a little more user-friendly, as long as I'm improving the feedback.